app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,110 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
```

Comment:
hmm... I think topic related things should be handled in the subclasses. If we use abstract topic which means we focus on the `Discussion::Topic` but there are only `actable_type` and `actable_id` inside it. It is pretty much like an interface.  Only when it is 'acted' by other actual classes, it is meaningful. And it's also meaningful to authorize or do other things...